worker/provisioner/provisioner_task.go

```diff
@@ -926,7 +928,12 @@ func (task *provisionerTask) setErrorStatus(message string, machine *apiprovisio
 		// Something is wrong with this machine, better report it back.
 		return errors.Annotatef(err2, "cannot set error status for machine %q", machine)
 	}
-	return err
+	return nil
+}
+
+func (task *provisionerTask) setErrorStatusAndRemoveMachineAZ(message string, machine *apiprovisioner.Machine, err error) error {
```

Comment:
this might be an unpopular opinion, but I find these sorts of do-two-unrelated-things functions unhelpful. I'd much rather see two consecutive function calls. alternatively, a function with a higher-level conceptual purpose, like "machineProvisioningFailed", which would do the things necessary to handle the failure

If it's handled all in one place (startMachines), then I wouldn't bother making a function for it

Reply:
I removed it,